### PR TITLE
MM-61225: Revert session pooling

### PR DESCRIPTION
**Pull request by agnivade**
This originated from https://github.com/mattermost/mattermost/issues/15249.

However, the original idea was discarded https://github.com/mattermost/mattermost/issues/15249#issuecomment-709713065
as being too complicated to implement. Then I had another
idea to implement it just for session objects.

My thinking was that since every single request allocates a new
session struct, it would be good to use a sync.Pool for that.

However, 4 years later, now we know that the primary bottleneck
in app performance comes from websocket event marshalling.
Therefore, while it would be good to do this, it is difficult
to do it correctly (as shown by the numerous racy tests).

Hence, reverting this.

```release-note
NONE
```

https://mattermost.atlassian.net/browse/MM-61225

### CODE REVIEW

**Comment by agnivade** [comment]
Also running a small load-test to spot check any major perf regression.

**Comment by agnivade** [comment]
Attaching the heap and CPU profiles from the test. I don't see any major allocations from Session objects. Yes, if you search by "Session" in the heap profile, you will see it, but it does not appear to be a major bottleneck compared to other things.
[session.zip](https://github.com/user-attachments/files/17486464/session.zip)

File: server/channels/app/platform/web_conn.go [hunk 1]
```diff
@@ -393,8 +388,6 @@ func (wc *WebConn) GetSession() *model.Session {
 
 // SetSession sets the session of the connection.
 func (wc *WebConn) SetSession(v *model.Session) {
-	// Clone the session first as WebConn takes ownership of the object
-	// and the web.Hub will return it to the [sync.Pool] once the WebConn gets removed.
 	if v != nil {
 		v = v.DeepCopy()
```
Comment (hanzei) [inline server/channels/app/platform/web_conn.go:392]: Do we need to copy the session here?

Comment (agnivade) [reply server/channels/app/platform/web_conn.go:392]: Yes, this is unrelated to the sync.Pool implementation: https://github.com/mattermost/mattermost/pull/16103. So it should still be there.

**Review by streamer45 — APPROVED** [review]
Good idea to revisit this now that we have enough data to prove it wasn't sufficiently worth the complexity :100:

**Review by hanzei — COMMENTED** [review]
Should we revert the changes from https://github.com/mattermost/mattermost/pull/28324 as part of this PR? They are no longer needed.

**Comment by agnivade** [comment]
@hanzei - PTAL.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Pull request merged**